meson.build

```diff
@@ -0,0 +1,153 @@
+  'enable-command-xxd':          '-DUSH_CONFIG_ENABLE_COMMAND_XXD',
+  'enable-command-ls':           '-DUSH_CONFIG_ENABLE_COMMAND_LS',
+
+  'enable-feature-commands':     '-DUSH_CONFIG_ENABLE_FEATURE_COMMANDS',
+  'enable-feature-autocomplete': '-DUSH_CONFIG_ENABLE_FEATURE_AUTOCOMPLETE',
+  'enable-feature-shell-styles': '-DUSH_CONFIG_ENABLE_FEATURE_SHELL_STYLES',
+}
+
+translation_map = {
+  'translation-ok':                        '-DUSH_CONFIG_TRANSLATION_OK',
+  'translation-error':                     '-DUSH_CONFIG_TRANSLATION_ERROR',
+  'translation-directory-not-found':       '-DUSH_CONFIG_TRANSLATION_DIRECTORY_NOT_FOUND',
+  'translation-nested-directories-exist':  '-DUSH_CONFIG_TRANSLATION_NESTED_DIRECTORIES_EXIST',
+  'translation-cannot-find-parent-node':   '-DUSH_CONFIG_TRANSLATION_CANNOT_FIND_PARENT_NODE',
+  'translation-directory-already-mounted': '-DUSH_CONFIG_TRANSLATION_DIRECTORY_ALREADY_MOUNTED',
+  'translation-syntax-error':              '-DUSH_CONFIG_TRANSLATION_SYNTAX_ERROR',
+  'translation-wrong-arguments':           '-DUSH_CONFIG_TRANSLATION_WRONG_ARGUMENTS',
+  'translation-file-not-executable':       '-DUSH_CONFIG_TRANSLATION_FILE_NOT_EXECUTABLE',
+  'translation-file-not-writable':         '-DUSH_CONFIG_TRANSLATION_FILE_NOT_WRITABLE',
+  'translation-file-not-readable':         '-DUSH_CONFIG_TRANSLATION_FILE_NOT_READABLE',
+  'translation-no-help-available':         '-DUSH_CONFIG_TRANSLATION_NO_HELP_AVAILABLE',
+  'translation-file-not-found':            '-DUSH_CONFIG_TRANSLATION_FILE_NOT_FOUND',
+  'translation-read-only-file':            '-DUSH_CONFIG_TRANSLATION_READ_ONLY_FILE',
+}
+
+foreach option_key, option_value : settings_map
+  add_project_arguments(option_value + '=' + get_option(option_key).to_string(), language: 'c')
+  summary(option_key, get_option(option_key), section: 'Enabled Settingss')
+endforeach
+
+foreach option_key, option_value : feature_map
+  if get_option(option_key)
+    add_project_arguments(option_value, language: 'c')
+  endif
+  summary(option_key, get_option(option_key), section: 'Enabled Features')
+endforeach
+
+foreach option_key, option_value : translation_map
+  add_project_arguments(option_value + '="' + get_option(option_key) + '"', language: 'c')
+endforeach
+
+if get_option('buildtype').startswith('debug')
+  add_project_arguments('-DDEBUG', language: 'c')
+  add_project_arguments('-ggdb3', language: 'c')
+endif
+
+# Target
+
+include_directories += public_headers
+
+project_target = static_library(
+  meson.project_name(),
+  project_source_files,
+  install: true,
+  c_args: build_args,
+  link_args: link_args,
+  include_directories: include_directories,
+  dependencies: project_dependencies,
+)
+
+# Project
+
+# Make this library usable as a Meson subproject
+project_dep = declare_dependency(
+  include_directories: public_headers,
+  link_with: project_target,
+)
+
+set_variable('lib' + meson.project_name() + '_dep', project_dep)
```

Comment:
This is very not-straightforward. Why are you using set_variable, instead of just assigning the name properly in the first place?

Reply:
I agree that it's not, but it allows for the prefix and suffix to be added to the project name. Might there be a better way?